tests/foreman/ui/test_discoveryrule.py

```diff
@@ -0,0 +1,168 @@
+            make_discoveryrule(session, name=name,
+                               hostgroup=self.hostgroup_name)
+            self.assertIsNotNone(self.discoveryrules.wait_until_element
+                                 (common_locators["name_haserror"]))
+            self.assertIsNone(self.discoveryrules.search(name))
+
+    @data("", "  ")
+    def test_negative_create_discovery_rule_2(self, name):
+        """@Test: Create Discovery Rule with blank and whitespace in name
+
+        @Feature: Foreman Discovery
+
+        @Assert: Error should be raised and rule should not be created
+
+        """
+        with Session(self.browser) as session:
+            make_discoveryrule(session, name=name,
+                               hostgroup=self.hostgroup_name)
+            self.assertIsNotNone(self.discoveryrules.wait_until_element
+                                 (common_locators["name_haserror"]))
+            self.assertIsNone(self.discoveryrules.search(name))
+
+    @data('-1', gen_string("alpha", 6))
+    def test_negative_create_discovery_rule_3(self, limit):
+        """@Test: Create Discovery Rule with invalid host limit
+
+        @Feature: Foreman Discovery
+
+        @Assert: Error should be raised and rule should not be created
+
+        """
+        name = gen_string("alpha", 6)
+        with Session(self.browser) as session:
+            make_discoveryrule(session, name=name,
+                               hostgroup=self.hostgroup_name, host_limit=limit)
+            self.assertIsNotNone(self.discoveryrules.wait_until_element
+                                 (common_locators["haserror"]))
+            self.assertIsNone(self.discoveryrules.search(name))
+
+    def test_negative_create_discovery_rule_4(self):
+        """@Test: Create Discovery Rule with name that already exists
+
+        @Feature: Foreman Discovery
+
+        @Assert: Error should be raised and rule should not be created
+
+        """
+        name = gen_string("alpha", 6)
+        with Session(self.browser) as session:
+            make_discoveryrule(session, name=name,
+                               hostgroup=self.hostgroup_name)
+            self.assertIsNotNone(self.discoveryrules.search(name))
+            make_discoveryrule(session, name=name,
+                               hostgroup=self.hostgroup_name)
+            self.assertIsNotNone(self.discoveryrules.wait_until_element
+                                 (common_locators["name_haserror"]))
+
+    def test_negative_create_discovery_rule_5(self):
+        """@Test: Create Discovery Rule with invalid priority
+
+        @Feature: Foreman Discovery
+
+        @Assert: Error should be raised and rule should not be created
+
+        """
+        name = gen_string("alpha", 6)
+        with Session(self.browser) as session:
+            make_discoveryrule(session, name=name,
+                               hostgroup=self.hostgroup_name,
+                               priority=gen_string("alpha", 6))
```

Comment:
That's real nitpick, but if you have time, please change double quotes to single one in all possible places